### PR TITLE
Add tracking to user interaction of deposits settings

### LINKED ISSUE #5882: Ability to measure view and user interaction with deposits settings
We need to be able to:
- Count how many times user view the deposits settings on the Payments > Settings page. If there is already a way to count the Payments > Settings view similar to Payments > Overview page view ([read](https://github.com/Automattic/woocommerce-payments/issues/5771#issuecomment-1468699248)) that might be enough.

- Count user's clicks on the "Learn more about pending schedules" link.

- Count user's interaction (on change, maybe?) with the deposit schedule dropdowns.

- Count user's clicks on the "Manage in Stripe" link.

Screenshot 2023-03-28 at 20 16 23

### PR DESCRIPTION
**Pull request by brucealdridge**
Fixes #5882 

#### Changes proposed in this Pull Request

Add tracking to user interactions to the deposits panel on the settings page.

Screenshot 2023-05-11 at 3 28 26 PM

When a Merchant clicks on the "Manage in Stripe" link to update account details, the event  `wcpay_settings_deposits_manage_in_stripe_click` will be tracked

The link-based events use an `onClick` which may not be the optimal event. In testing, this works with both mouse, keyboard, and touch navigation.

#### Testing instructions

Visit the WCPay Settings Screen, scroll down to the deposits section.

* Click the Manage in Stripe link
* See event `wcpay_settings_deposits_manage_in_stripe_click` has been tracked

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6037` or branch name `add/5882-tracks-for-deposit-settings` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=add/5882-tracks-for-deposit-settings) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 8c814fbfc77fce9fc5e42fd37c3db3408d8ed30c
- Build time: 2023-05-12 11:53:04 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/settings/deposits/index.js [hunk 1]
```diff
@@ -162,6 +180,12 @@ const DepositsSchedule = () => {
 					href="https://woocommerce.com/document/payments/faq/deposit-schedule/"
 					target="_blank"
 					rel="external noreferrer noopener"
+					onClick={ () =>
+						wcpayTracks.recordEvent(
+							'wcpay_deposits_schedule_disabled_help',
```
Comment (haszari) [inline client/settings/deposits/index.js]: Quick feedback on the event naming (assuming this is still in progress):

- Use a consistent prefix for the event source, i.e. where events are triggered. Suggest `wcpay_settings_deposits_`.
- Name the action similar to the UI element – we are tracking the user clicking or viewing something; _what_ is the thing being clicked?
- Use an action suffix, e.g. `view` or `click`.

So for this example, I'd suggest something like: `wcpay_settings_deposits_new_account_notice_help_click`, similar to proposed [`wcpay_overview_deposits_new_account_notice_docs_click`](https://github.com/Automattic/woocommerce-payments/issues/5881#issuecomment-1530793008).

Though … we might not need this event yet (I didn't add that event to overview page).

File: client/settings/deposits/index.js [hunk 2]
```diff
@@ -59,6 +61,22 @@ const CustomizeDepositSchedule = () => {
...
+		wcpayTracks.recordEvent( 'wcpay_deposits_schedule_change', {
+			interval: depositScheduleInterval,
+			weekly_anchor: depositScheduleWeeklyAnchor,
+			monthly_anchor: depositScheduleMonthlyAnchor,
+		} );
+	}, [
+		depositScheduleInterval,
+		depositScheduleMonthlyAnchor,
+		depositScheduleWeeklyAnchor,
+	] );
```
Comment (haszari) [inline client/settings/deposits/index.js]: This event looks like a `wcpay_settings_deposits_view` event – it fires when the settings UI is rendered.

Assuming that's the case … do we need a custom event? i.e. maybe there's a wcadmin event we can use. I see `wcadmin_settings_view` and `wcadmin_page_view` firing when loading this page.

If we do need a custom `view` event here:

- Name needs to accurately reflect action, e.g. `wcadmin_settings_payments_deposits_view`.
- I don't think it should have props for the current settings – it hasn't changed yet.
  - We could add an event for logging changes to schedule, e.g. `wcpay_settings_deposit_schedule_change`. This might need to be fired from server where we can detect the settings have changed.
- The `useInitialRenderEffect` hook pattern could be a custom hook, or even better, add a named `useRenderTracksEvent` hook so we can consistently use it.
- View events are challenging here as the UI is off-screen at the start, so me might need a smarter event (that understands scroll position). 

TLDR let's use `wcadmin_settings_view` for now :)

File: client/settings/deposits/index.js (hunk 1 above)
Comment (brucealdridge) [reply client/settings/deposits/index.js]: > Quick feedback on the event naming (assuming this is still in progress)

It was still in progress and has now been updated.

File: client/settings/deposits/index.js (hunk 2 above)
Comment (brucealdridge) [reply client/settings/deposits/index.js]: > This event looks like a wcpay_settings_deposits_view event – it fires when the settings UI is rendered.

The intention of this was to only fire this event when the schedule was changed, not on render/view. In my testing, this only fired when the schedule was changed.

> This might need to be fired from server where we can detect the settings have changed.

We already track this on server. Even though the issue mentions tracking on change, there doesn't appear to be a need for this additional tracking so I have removed.

Comment (haszari) [reply client/settings/deposits/index.js]: Noting the settings change event: `wcpay_deposit_schedule_change`

File: changelog/add-5882-tracks-for-deposit-settings [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: dev
+
+Add tracks events to deposit settings
```
Comment (haszari) [inline changelog/add-5882-tracks-for-deposit-settings]: I think we should leave this empty so all the tracks changes fall under one changelog ([there's a generic entry in this PR](https://github.com/Automattic/woocommerce-payments/pull/6188/files)).

TIL that it's ok to leave the string blank after selecting `patch`, although the tool strongly encourages adding an entry!

File: client/settings/settings-manager/index.js [hunk 4]
```diff
@@ -100,7 +101,17 @@ const DepositsDescription = () => {
...
 				) }
 			</p>
-			<ExternalLink href="https://woocommerce.com/document/payments/faq/deposit-schedule/#section-2">
+			<ExternalLink
+				href="https://woocommerce.com/document/payments/faq/deposit-schedule/#section-2"
+				onClick={ () =>
+					wcpayTracks.recordEvent(
+						wcpayTracks.events
+							.SETTINGS_DEPOSITS_PENDING_SCHEDULES_DOCS_CLICK,
+						'wcpay_deposits_pending_schedules_docs_click',
```
Comment (haszari) [inline client/settings/settings-manager/index.js]: I couldn't get this to fire – does `ExternalLink` support an `onClick` handler?

**Review by haszari — COMMENTED** [review]
#### TLDR
- I don't think we need the notice events right now – can remove. 
- "Deposit schedule change" and "edit bank ac in stripe" events are working well.
- Pending period / `ExternalLink` event wasn't firing for me (but maybe not needed).

---
This is looking good overall, however I don't think we need to track all the docs links. I decided against this in [deposits list PR](https://github.com/Automattic/woocommerce-payments/pull/6188), though we should check with product – if any of these are important we can add tracks events as needed or consistently for all. Also note for these notices we could ship a `view` event and a `click` event potentially. [See the deposits list issue for examples and discussion](https://github.com/Automattic/woocommerce-payments/issues/5881#issuecomment-1530793008).

It doesn't hurt to add these events but at the same time I think we should consistently track the most important interactions only.

#### Events
Here's a list of all relevant events, what I tested, and what I think we should ship in this version (at least until we get more info from product).

| Event | Test result | Needed? |
| ----- | ----------- | ------- |
| `wcadmin_wcpay_settings_deposits_manage_in_stripe_click` | Confirmed ✅  | 🚢 IMO important to track | 
| `wcadmin_wcpay_settings_deposits_waiting_period_docs_click` | Confirmed ✅  | IMO not necessary unless we decide to track notices/notice clicks | 
| `wcadmin_wcpay_settings_deposits_pending_schedules_docs_click ` | 🔴 didn't fire for me  | IMO not necessary unless we decide to track notices/notice clicks | 
| `wcadmin_wcpay_settings_deposits_schedule_disabled_docs_click` | ❓ don't know how to get that notice | IMO not necessary unless we decide to track notices/notice clicks | 
| `wpcom_wcpay_deposit_schedule_change` | Confirmed ✅  | 🚢 IMO important to track | 

Testing notes:

- For most events, I tested in Chrome with dev tools to confirm event is sent (`t.gif` request).
- For `wpcom_wcpay_deposit_schedule_change` I didn't test directly (my store is dev mode, stuck in waiting period). But I saw the event is firing consistently in tracks backend.
- `wcpay_settings_deposits_pending_schedules_docs_click` – if we do need to track this event, we'll need to figure out why it's not firing (`ExternalLink`?). Otherwise we can remove the event if not important for now.

> The link-based events use an onClick which may not be the optimal event. In my testing, this works with both mouse and keyboard navigation.


- I tested `wcadmin_wcpay_settings_deposits_manage_in_stripe_click` on mobile safari (touch event) and confirmed it showed up in tracks live view (~30 mins later). So using `onClick` is good to go IMO :)

File: client/settings/settings-manager/index.js (hunk 4 above)
Comment (brucealdridge) [reply client/settings/settings-manager/index.js]: I've re-tested and it's not firing for me now either but am confident it was working. The `SETTINGS_DEPOSITS_MANAGE_IN_STRIPE_CLICK` is also using an `ExternalLink` with an `onClick` event and is firing. 
I'm not sure why one is working and the other isn't.

File: changelog/add-5882-tracks-for-deposit-settings (hunk 3 above)
Comment (brucealdridge) [reply changelog/add-5882-tracks-for-deposit-settings]: I've removed this changelog entry

File: client/settings/settings-manager/index.js (hunk 4 above)
Comment (brucealdridge) [reply client/settings/settings-manager/index.js]: I've removed this event as I am unable to resolve this to give consistent tracking.

**Comment by haszari** [comment]
> I don't think we need to track all the docs links. I decided against this in https://github.com/Automattic/woocommerce-payments/pull/6188, though we should check with product – if any of these are important we can add tracks events as needed or consistently for all. 

@brucealdridge  what do you think about removing the notice click events? I don't think they are essential.

**Review by shendy-a8c — APPROVED** [review]
Tested well and looks good.

Reminder to register the new event at. Read PCYsg-mUj-p2.

**Comment by shendy-a8c** [comment]
Will merge to make it to the code freeze.
